### DISCUSSION TITLE
Intelligent Octopus and Zappi charger

### DISCUSSION THREAD

**Discussion by nigels0** [BottlecapDave/HomeAssistant-OctopusEnergy#243] [Ideas]
The combo of IO and Zappi charger is quite common, but to use this efficiently with IO, the charger has to be put into dumb mode (Fast) for Octopus to charge through to the car (in my case, I have a VW ID.5), but during the day, the charger should be on Eco+ mode. 

It’s possible with the Myenergi API and integration for HA to switch the Zappi from Fast to Eco+, but it isn’t currently possible (as far as I’m aware) to switch smart charging on or off except via the app.

it would be ideal to be able to automatically do this in HA by switching the smart charging option in IO on and Zappi to Fast during off peak times and off during peak times where the charger would use any excess from solar panels.  Currently, I have to do this manually in the app.

Not sure whether this is an item for the integration - it may not be available via the Octopus API

**Comment by BottlecapDave** [comment]
Hi there. I'm not only intelligent so only have experience second hand. There is a separate integration that provides this functionality. However, there have been [ongoing discussions](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/114) about bringing this functionality to this integration (including discussions with the developer of the other integration). The build that is advertised is a little old, but I didn't want to include the functionality until I got the basics correct but I'm awaiting confirmation. If you can help review/confirm the advertised build in the thread, then I can look at bringing the other functionality (like turning smart charging on/off)

**Reply by nigels0** [reply]
Thanks. I actually got this working with a combination of those integrations and an automation. 

The two extra integrations are Octopus Intelligent Tariff and, of course, the Myenergi one. 

The Tariff integration does indeed allow you to switch smart charging on and off.

I switch it off after sunrise + 2 hours and set the Zappi to Eco+, then revert to On and Fast during off peak times. 

I’m using the binary_sensor.octopus_intelligent_slot entity to manage on and off times as There may be other times when Octopus shifts to off-peak outside of the usual 23:30 to 05:30.
